pkg/gateway/model_build_targetgroup.go

```diff
@@ -192,23 +193,39 @@ func (t *latticeServiceModelBuildTask) buildTargets(ctx context.Context) error {
 // Build target group for K8S serviceexport object
 func (t *targetGroupModelBuildTask) BuildTargetGroup(ctx context.Context) error {
 	tgName := latticestore.TargetGroupName(t.serviceExport.Name, t.serviceExport.Namespace)
+	var tg *latticemodel.TargetGroup
+	var err error
+	if t.serviceExport.DeletionTimestamp.IsZero() {
+		// build TargetGroup Model for serviceExport creation
+		tg, err = t.buildTargetGroupForServiceExportCreation(ctx, tgName)
+	} else {
+		// build TargetGroup Model for serviceExport deletion
+		tg, err = t.buildTargetGroupForServiceExportDeletion(ctx, tgName)
+	}
+	if err != nil {
+		return err
+	}
+	t.tgByResID[tgName] = tg
+	t.targetGroup = tg
+	return nil
+}
 
+func (t *targetGroupModelBuildTask) buildTargetGroupForServiceExportCreation(ctx context.Context, targetGroupName string) (*latticemodel.TargetGroup, error) {
 	svc := &corev1.Service{}
 	if err := t.client.Get(ctx, k8s.NamespacedName(t.serviceExport), svc); err != nil {
-		// mark there is no serviceexport dependence on the TG
-		t.datastore.SetTargetGroupByServiceExport(tgName, false, false)
-		return fmt.Errorf("error finding corresponding service %v error :%w", k8s.NamespacedName(t.serviceExport), err)
+		t.log.Infof("Error finding corresponding service %v error :%v \n", k8s.NamespacedName(t.serviceExport), err)
+		t.datastore.SetTargetGroupByServiceExport(targetGroupName, false, false)
+		return nil, err
```

Comment:
I do `not` think this is right.  If there is no serviceexport, then the creation should fail

Reply:
```
svc := &corev1.Service{}
if err := t.client.Get(ctx, k8s.NamespacedName(t.serviceExport), svc); err != nil {
```
it tried to get service, not serviceexport..

If there is no serviceexport, then the creation indeed fail at [serviceexport_controller.go line 128](https://github.com/aws/aws-application-networking-k8s/blob/786b9efb0232675daa85373012d5899a1204f8a7/controllers/serviceexport_controller.go#L128)

If there is no service,  the creation also fail, in here at line 218, it `return nil, err`,

Reply:
Today, if there is no K8S service, we do `not` create target group.  Why change this behavior

Reply:
I think both old and new behavior are all:  "if there is no K8S service, we do not create target group", Which code change you suspect it changed this behaviour?

Reply:
+1 to @zijun726911 here, I don't see any logical change in this part at least.